### PR TITLE
Feat/proposal fetching strategy improvement

### PR DESCRIPTION
**Pull request by mstrasinskis**
# Motivation

To avoid too many request (especially update) the queryAndUpdate `strategy` was implemented.

Request flow changes:
- Voting History / fetching ballot proposal - query only
- reload proposals after voting - update only

# Changes

- add QueryAndUpdateStrategy parameter
- All history-view related proposal calls use query
- after voting the proposal is updated with the `update` call. It is longer but we avoid some temporary states because the query could not contain updated data (at least on testnet)

# Tests

- strategy tests

### CODE REVIEW

File: frontend/svelte/src/lib/components/proposal-detail/VotingCard/VotingCard.svelte [hunk 1]
```diff
@@ -24,9 +24,10 @@
   /** Signals that the initial checkbox preselection was done. To avoid removing of user selection after second queryAndUpdate callback. */
   let initialSelectionDone = false;
 
-  $: visible =
-    votableNeurons().length > 0 &&
-    proposalInfo.status === ProposalStatus.PROPOSAL_STATUS_OPEN;
+  $: $neuronsStore,
```
Comment (mstrasinskis) [inline frontend/svelte/src/lib/components/proposal-detail/VotingCard/VotingCard.svelte:27]: tiny not important visual fix

**Review by mstrasinskis — COMMENTED** [review]
@peterpeterparker All the queries were reset to "query_and_update" except one - https://github.com/dfinity/nns-dapp/pull/717/files#diff-60c180d8fc2c4be28fe38b6f33fcf692ffa3467223282115a198ec5a537c02b7R326

File: frontend/svelte/src/lib/components/neuron-detail/Ballots/BallotSummary.svelte [hunk 2]
```diff
@@ -17,6 +17,7 @@
         proposalId: ballot.proposalId as ProposalId,
         setProposal: (proposalInfo: ProposalInfo) => (proposal = proposalInfo),
         silentErrorMessages: true,
+        // TODO: think about strategy: "query"
```
Comment (peterpeterparker) [inline frontend/svelte/src/lib/components/neuron-detail/Ballots/BallotSummary.svelte:20]: should we open a jira task and add the ref in the comment?

File: frontend/svelte/src/lib/services/utils.services.ts [hunk 3]
```diff
@@ -12,51 +12,71 @@ export type QueryAndUpdateOnError<E> = (options: {
...
-  logWithTimestamp(`${logPrefix} calls...`);
-
-  return Promise.race([
-    // query
+  const query = () =>
     request({ certified: false, identity })
       .then((response) => {
-        logWithTimestamp(`${logPrefix} query complete.`);
         if (certifiedDone) return;
         onLoad({ certified: false, response });
```
Comment (peterpeterparker) [inline frontend/svelte/src/lib/services/utils.services.ts:51]: in `onLoad` and `onError` we can maybe spare something?

is

```
onLoad({ certified: false, response });
onError?.({ certified: false, error });
```

can't it be

```
onLoad({ certified, response });
onError?.({ certified, error });
```

**Review by peterpeterparker — COMMENTED** [review]
two small questions

File: frontend/svelte/src/lib/components/neuron-detail/Ballots/BallotSummary.svelte (hunk 2 above)
Comment (mstrasinskis) [reply frontend/svelte/src/lib/components/neuron-detail/Ballots/BallotSummary.svelte:20]: Good point, done.

File: frontend/svelte/src/lib/services/utils.services.ts (hunk 3 above)
Comment (mstrasinskis) [reply frontend/svelte/src/lib/services/utils.services.ts:51]: Refactored.

**Review by peterpeterparker — APPROVED** [review]
LGTM, thanks

**Pull request merged**